docs/source/configuration.rst

```diff
@@ -99,4 +99,52 @@ RESPONSE_VALIDATION
     that they conform to the protocol. This should only be used for development
     purposes.
 
+OIDC_PROVIDER
+    If this value is provided, then OIDC is configured and SSL is used. It is
+    the URI of the OpenID Connect provider, which should return an OIDC
+    provider configuration document.
 
+SERVER_NAME
+    A standard flask configuration variable, but required if OIDC is configured
+    so the server can know its correct redirect URI.
+
+OIDC_CLIENT_ID, OIDC_CLIENT_SECRET
+    These are the client id and secret arranged with the OIDC provider,
+    if client registration is manual (google, for instance). If the provider
+    supports automated registration they are not required or used.
```

Comment:
I think we should discourage dynamic registration in general.  That's been Knox and my suggestion to the SWG to put into a "best practices" document.  The reason is that multiple processes registering themselves with an OP will result in multiple different shared keys unless extreme care is taken to synchronize the RP/OP shared secret among all instances of the RP.  Or, if an OP is poorly configured and is spanning more than one process space you can have the same problem, ie. lack of synchronization of secrets.

This server is pretty stateless and I think it's important to keep it that way.  OAuth2/OIDC are insidious in their introduction of hidden state, and this is one of those features where it creeps in.